stackslib/src/net/api/gethealth.rs

```diff
@@ -156,97 +102,26 @@ impl RPCRequestHandler for RPCGetHealthRequestHandler {
-                    );
+        // There could be a edge case where our node is ahead of all peers.
+        let difference_from_max_peer =
+            max_stacks_height_of_neighbors.saturating_sub(node_stacks_tip_height);
 
-                    // get max block height amongst bootstrap nodes
-                    match network.inv_state.as_ref() {
-                        Some(inv_state) => {
-                            inv_state.get_max_stacks_height_of_neighbors(neighbors_arg.as_deref(), ibd)
-                        }
-                        None => {
-                            return create_error_response(
-                                &preamble,
-                                "Peer inventory state (Epoch 2.x) not found, unable to determine health.",
-                            );
-                        }
-                    }
-                } else {
-                    let neighbors_arg: Option<Vec<NeighborAddress>> = neighbors_arg.as_ref().map(|v| v.iter().map(NeighborAddress::from_neighbor).collect());
-                    match network.block_downloader_nakamoto.as_ref() {
-                        Some(block_downloader_nakamoto) => {
-                            block_downloader_nakamoto.get_max_stacks_height_of_neighbors(neighbors_arg.as_deref())
-                        }
-                        None => {
-                            return create_error_response(
-                                &preamble,
-                                "Nakamoto block downloader not found (Epoch 3.0+), unable to determine health.",
-                            );
-                        }
-                    }
-                }
-            };
-
-            match peer_max_stacks_height_opt {
-                Some(max_stacks_height_of_neighbors) => {
-                    // There could be a edge case where our node is ahead of all peers.
-                    let node_stacks_tip_height = network.stacks_tip.height;
-                    let difference_from_max_peer =
-                        max_stacks_height_of_neighbors.saturating_sub(node_stacks_tip_height);
-
-                    let preamble = HttpResponsePreamble::ok_json(&preamble);
-                    let data = RPCGetHealthResponse {
-                        difference_from_max_peer,
-                        max_stacks_height_of_neighbors,
-                        node_stacks_tip_height,
-                    };
-                    let body = HttpResponseContents::try_from_json(&data)?;
-                    Ok((preamble, body))
-                }
-                None => create_error_response(
-                    &preamble,
-                    "Couldn't obtain stats on any bootstrap peers, unable to determine health.",
-                ),
-            }
-        })
+        let preamble = HttpResponsePreamble::ok_json(&preamble);
+        let data = RPCGetHealthResponse {
+            difference_from_max_peer,
+            max_stacks_height_of_neighbors,
```

Comment:
Do you think you could also track and report the IP and port number of the peer which reported the max height?  This would be very useful when diagnosing why the local view of the maximum Stacks block height is off.